### ISSUE TITLE
When multi-targeting, Analyzer reports problems that cannot be fixed in a prior version of xUnit

### ISSUE THREAD

**Issue by NightOwl888** [xunit/xunit#1426]
When creating a multi-targeted test project, I am referencing older versions of xUnit so I can test on .NET 3.5 and 4.0 in addition to .NET Core 1.0 and 2.0. 

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>netcoreapp2.0;netcoreapp1.0;net40;net35</TargetFrameworks>
    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">$(PackageTargetFallback);dnxcore50</PackageTargetFallback>

    <!-- For now, we can switch framework to test by manually editing this file -->
    <!--<TargetFramework>net35</TargetFramework>-->
    <Version>0.4.1</Version>
    <AssemblyName>Spatial4n.Tests</AssemblyName>
    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
    <NoWarn>1591</NoWarn>
    <GenerateAssemblyFileVersionAttribute>false</GenerateAssemblyFileVersionAttribute>
    <DefineConstants>FEATURE_STRONG_NAME</DefineConstants>
    <FrameworkPathOverride Condition="'$(TargetFramework)' == 'net35'">$(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\Framework\.NETFramework\v3.5\Profile\Client</FrameworkPathOverride>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="..\CommonAssemblyInfo.cs" Link="Properties\CommonAssemblyInfo.cs" />
  </ItemGroup>

  <PropertyGroup>
    <DefineConstants>$(DefineConstants);FEATURE_NTS</DefineConstants>
  </PropertyGroup>
  
  <PropertyGroup Condition="$(DefineConstants.Contains('FEATURE_STRONG_NAME'))">
    <SignAssembly>True</SignAssembly>
    <AssemblyOriginatorKeyFile>..\Spatial4n.snk</AssemblyOriginatorKeyFile>
  </PropertyGroup>

  <PropertyGroup>
    <NoWarn>$(NoWarn);1701;1702;1705</NoWarn>
  </PropertyGroup>

  <PropertyGroup Condition=" $(TargetFramework.StartsWith('netcore')) ">
    <DefineConstants>$(DefineConstants);NETCOREAPP</DefineConstants>
  </PropertyGroup>
  
  <ItemGroup Condition="!$(TargetFramework.StartsWith('netcore'))">
    <PackageReference Include="xunit" Version="1.9.2" />
    <PackageReference Include="xunit.extensions" Version="1.9.2" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.1.0" />
  </ItemGroup>
  
  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
    <PackageReference Include="xunit" Version="2.2.0" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
  </ItemGroup>

  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">
    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.0-beta4-build3742" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
    <PackageReference Include="xunit" Version="2.3.0-beta4-build3742" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.0-beta4-build3742" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\Spatial4n.Core.NTS\Spatial4n.Core.NTS.csproj" />
    <ProjectReference Include="..\Spatial4n.Core\Spatial4n.Core.csproj" />
  </ItemGroup>

  <ItemGroup>
    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
  </ItemGroup>

</Project>
```

xUnit complains about Assert.Equal being used with a string.

```
Assert.Equal(string1, string2);
```

In .NET Core, I can fix that by changing to:

```
Assert.Equal(string1, string2, false);
```

However, this does not compile on xUnit 1.9.2 or 2.2.0 because the overloads do not exist. So, I put in some branching logic:

```
#if NETCOREAPP
    Assert.Equal(string1, string2, false);
#else
    Assert.Equal(string1, string2);
#endif
```

Ugly, but it compiles. However, the second assert (for net35 and net40) triggers the warning not to use Assert.Equal's generic overload with strings. But there is no way to fix it in that case because the overload doesn't exist in xUnit 1.9.2.

The only solution for this seems to be:

```
#if NETCOREAPP
    Assert.Equal(string1, string2, false);
#else
#pragma warning disable xUnit2006
    Assert.Equal(string1, string2);
#pragma warning restore xUnit2006
#endif
```

Although I can't be sure because the [analyzers indicate it is wrong, but don't provide any solutions as to how to fix the problem](https://github.com/xunit/xunit/issues/1423).

If there is a better solution for fixing this, please inform me. For now, I am just ignoring `xUnit2006` project-wide because I don't want to put this ridiculous branching logic everywhere in my code for a problem that can't be fixed anyway.

**Comment by marcind** [comment]
There's a lot of pieces in this issue, but my initial reaction is that I'm surprised things have been working for you so far targeting xunit 1.9 and 2.x from the same source, 2.0 being a complete rewrite etc.

At this point I'm not sure the analyzers will ever explicitly support being executed against 1.x-based tests, but it's also unclear to me how they end up being executed since they're only part of 2.3-beta2 and above.

As to the analyzer causing a failure against 2.2, that's a bug. The analyzers are intended to target 2.0 and above and smartly detect if a feature is not supported in an earlier version. Once this gets fixed the 1.9 issue might also be resolved, although that would be coincidental and not intentional.

**Comment by marcind** [comment]
Closing this as multi-targetting across xunit versions is not a supported goal of the analyzers.
